Mathlib/Analysis/SpecialFunctions/Trigonometric/Inverse.lean

```diff
@@ -363,6 +364,9 @@ theorem arccos_cos {x : ℝ} (hx₁ : 0 ≤ x) (hx₂ : x ≤ π) : arccos (cos
   rw [arccos, ← sin_pi_div_two_sub, arcsin_sin] <;> simp [sub_eq_add_neg] <;> linarith
 #align real.arccos_cos Real.arccos_cos
 
+lemma arccos_eq_of_eq_cos (hy₀ : 0 ≤ y) (hy₁ : y ≤ π) (hxy : x = cos y) : arccos x = y := by
```

Comment:
This lemma feels contrived to me, as all lemmas with `(hxy : x = ..)` do. To make it worthwhile, is it true as an `iff`?

Reply:
I think this lemma deserves to exist since the entire point of `arccos` is that it's the inverse of `cos`. In my case, I would have to do
```
lemma angle_eq_abs_arg (hx : x ≠ 0) (hy : y ≠ 0) : angle x y = |(x / y).arg| := by
  suffices : ⟪x, y⟫_ℝ / (‖x‖ * ‖y‖) = Real.cos |arg (x / y)|
  · rw [angle, this, Real.arccos_cos (abs_nonneg _) (abs_arg_le_pi _)]
  rw [Real.cos_abs , Complex.cos_arg (div_ne_zero hx hy)]
  have := (map_ne_zero Complex.abs).2 hx
  have := (map_ne_zero Complex.abs).2 hy
  simp [div_eq_mul_inv, Complex.normSq_eq_norm_sq]
  field_simp
  ring
```
which is IMO much less natural and forces me to write the LHS and RHS a second time.

Reply:
Note that backticks on Github need ` ```lean `

Reply:
Note that you can edit my comments.

Reply:
@eric-wieser what do you think here?